crates/full-node/sov-ethereum/src/handlers/mod.rs

```diff
@@ -147,17 +151,47 @@ where
     let (tx_hash, raw_message) = ethereum
         .make_raw_tx(raw_evm_tx)
         .map_err(|e| to_jsonrpsee_error_object(e, ETH_RPC_ERROR))?;
-
     let tx = Seq::Rt::encode_with_ethereum_auth(RawTx::new(raw_message));
+    let mut state = ethereum.sequencer.api_state().default_api_state_accessor().to_provable_reader();
+    let (_decoded_tx, auth_data, _call) = <Seq::Rt as Runtime<S>>::Auth::authenticate(&tx, &mut state).map_err(|e| to_jsonrpsee_error_object(format!("Authentication failed: {}", e), ETH_RPC_ERROR))?;
+    let mut state = state.api_state_accessor;
+    let AuthorizationData { credential_id, uniqueness, .. } = auth_data;
+    drop(auth_data); // Drop auth_data to avoid holding it across the `await` point. This is required because it contains an `rc` which isn't thread safe.
+    for _ in 0..20 {
+        match uniqueness {
+            UniquenessData::Nonce(nonce) => {
+                let expected_nonce = sov_uniqueness::Uniqueness::<S>::default().nonce(&credential_id, &mut state)?.unwrap_or_default();
+                if nonce == expected_nonce {
+                    ethereum.sequencer.accept_tx(tx).await.map_err(|e| {
+                        to_jsonrpsee_error_object(
+                            format!("{} - '{}' ({:?})", e.status, e.message, e.details),
+                            ETH_RPC_ERROR,
+                        )
+                    })?;
+                    return Ok(tx_hash);
+                } else if nonce < expected_nonce {
```

Comment:
We could also add another condition:` if nonce > expected_nonce + 100, return an error`. This indicates that the nonce is too far in the future, so we won't wait for it.